File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        def find(node, level):
            if node.left is None and node.right is None:
                nonlocal result
                result = min(result, level)
                return

            if node.left:
                find(node.left, level + 1)
            if node.right:
                find(node.right, level + 1)


        if root is None:
            return 0

        result = int(1e9)
        find(root, 1)
        return result

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.cpp
class Solution {
private:
    int result;

    void find(TreeNode* node, int level) {
        if (node->left == NULL && node->right == NULL) {
            result = min(result, level);
            return;
        }

        if (node->left != NULL) {
            find(node->left, level + 1);
        }

        if (node->right != NULL) {
            find(node->right, level + 1);
        }
    }

public:
    int minDepth(TreeNode* root) {
        if (root == NULL) {
            return 0;
        }

        result = INT_MAX;
        find(root, 1);
        return result;
    }
};